### PR TITLE
Pipe prototypes

### PR DESCRIPTION
**Pull request by collinlunn**
Prototype for each basic pipe on the ground, rotatable pipes in spawn menu for mapping.

### CODE REVIEW

File: Resources/Prototypes/Entities/Constructible/Ground/pipes.yml [hunk 1]
```diff
@@ -20,26 +22,143 @@
...
+  id: PipeSEBend
+  suffix: SEBend
+  components:
+  - type: Icon
+    texture: Constructible/Atmos/pipe.rsi/pipeSEBend2.png
+  - type: NodeContainer
+    nodes:
+    - !type:PipeNode
+      nodeGroupID: Pipe
+      pipeDirection: SEBend
```
Comment (Zumorica) [inline Resources/Prototypes/Entities/Constructible/Ground/pipes.yml:99]: I wonder if it'd be best to just have a single bent pipe and simply change rotation with `RotatableComponent` or something similar.

File: Resources/Prototypes/Entities/Constructible/Ground/pipes.yml [hunk 2]
```diff
@@ -20,26 +22,143 @@
...
+  id: PipeTEast
+  suffix: TEast
+  components:
+  - type: Icon
+    texture: Constructible/Atmos/pipe.rsi/pipeTEast2.png
+  - type: NodeContainer
+    nodes:
+    - !type:PipeNode
+      nodeGroupID: Pipe
+      pipeDirection: TEast
```
Comment (Zumorica) [inline Resources/Prototypes/Entities/Constructible/Ground/pipes.yml:151]: Same with these, hmm.

**Review by Zumorica — COMMENTED** [review]
This reminds me heavily of edge firelocks, since those had to change airflow block direction when rotated. Maybe a similar thing could be done with these pipes? Other than that, this seems fine.

**Comment by collinlunn** [comment]
Adjusting pipeDirection from a rotation doesn't look like it will be that hard, I will look into it.

**Comment by Zumorica** [comment]
Cool, thanks! There's a `RotateEvent` that gets called when rotation is changed now, so it should make things much easier. Also feel free to look at `AirtightComponent` to see how I fixed a similar issue (the edge firelock thing I mentioned)

**Comment by collinlunn** [comment]
There is now just one of each pipe shape, and you can rotate it. One issue is that the green highlighted preview of what you are about to spawn in doesn't rotate, which it does for edge firelocks, and I'm not sure what the cause is.

**Comment by Zumorica** [comment]
Yeah, that's quite odd. I'll take a look at it later and see if I can figure out the reason hmm

**Comment by collinlunn** [comment]
May be related to the visualizers pipes use. The preview is properly getting what the pipe should look like given its yaml data, but does not update when its appearance is further modified in Initialize and beyond when rotations are applied.

**Comment by PJB3005** [comment]
Conflicts

**Comment by collinlunn** [comment]
I've fixed the spawn menu preview and it rotates properly for pipes.

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**